### PR TITLE
APL-2118 Refactor deploy container job to be able to trigger it manually on failure

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?
Refactor a bit deploy containers job, to use child pipelines. To be able to trigger deploy container jobs manually when there is a failure somewhere in the RC pipelines

Example pipeline: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/21215171

### Motivation

Be able to deploy docker image even when there is a failure in a RC pipeline. For now if one job fails (kitchen test for example) during the RC pipeline, the docker images will not be deployed and it is very painful to deploy them manually

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]
LGTM for ASC

File: tasks/agent.py [hunk 1]
```diff
@@ -855,7 +855,9 @@ def clean(ctx):
 
 
 @task
-def version(ctx, url_safe=False, omnibus_format=False, git_sha_length=7, major_version='7', version_cached=False):
+def version(
+    ctx, url_safe=False, omnibus_format=False, git_sha_length=7, major_version='7', version_cached=False, pipeline_id=""
```
Comment (chouquette) [inline tasks/agent.py:859]: nit: Wouldn't `None` be more appropriate for `pipeline_id` default value?

File: tasks/utils.py [hunk 2]
```diff
@@ -366,10 +366,18 @@ def cache_version(ctx, git_sha_length=7, prefix=None):
...
 def get_version(
-    ctx, include_git=False, url_safe=False, git_sha_length=7, prefix=None, major_version='7', include_pipeline_id=False
+    ctx,
+    include_git=False,
+    url_safe=False,
+    git_sha_length=7,
+    prefix=None,
+    major_version='7',
+    include_pipeline_id=False,
+    pipeline_id="",
```
Comment (chouquette) [inline tasks/utils.py:376]: ditto for `pipeline_id`

File: .gitlab/deploy_containers/conditions.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+
+
+.manual_auto_on_rc:
```
Comment (chouquette) [inline .gitlab/deploy_containers/conditions.yml:3]: Could you add a comment to clarify the intent of this rule?

File: .gitlab/deploy_containers/conditions.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    allow_failure: true
+    variables:
+      AGENT_REPOSITORY: agent
+      DSD_REPOSITORY: dogstatsd
+      IMG_REGISTRIES: public
+  - when: manual
+    allow_failure: true
+    variables:
+      AGENT_REPOSITORY: agent
+      IMG_REGISTRIES: public
```
Comment (chouquette) [inline .gitlab/deploy_containers/conditions.yml:28]: Shouldn't this also specify `DSD_REPOSITORY`?

**Review by AliDatadog — APPROVED** [review]

File: tasks/agent.py (hunk 1 above)
Comment (KevinFairise2) [reply tasks/agent.py:859]: Yes I agree!

File: tasks/utils.py (hunk 2 above)
Comment (KevinFairise2) [reply tasks/utils.py:376]: Updated

File: .gitlab/deploy_containers/conditions.yml (hunk 4 above)
Comment (KevinFairise2) [reply .gitlab/deploy_containers/conditions.yml:28]: I think it should

File: .gitlab/deploy_containers/conditions.yml (hunk 3 above)
Comment (KevinFairise2) [reply .gitlab/deploy_containers/conditions.yml:3]: Added some comments

File: .gitlab/deploy_containers/conditions.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
+
+# Rule to make job manual when running deploy pipelines, or automatic and on success on RC pipelines
+.manual_auto_on_rc:
```
Comment (chouetz) [inline .gitlab/deploy_containers/conditions.yml:3]:
```suggestion
.manual_on_deploy_or_auto_on_rc:
```
It's maybe more explicit like this as you don't have the comment where the rule is used

File: .gitlab/deploy_containers/conditions.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  - if: $BUCKET_BRANCH != "beta" && $BUCKET_BRANCH != "stable"
```
Comment (chouetz) [inline .gitlab/deploy_containers/conditions.yml:4]: So `nightly` fails here, is it expected?

**Review by chouetz — APPROVED** [review]
Few comments, up to you

File: .gitlab/deploy_containers/conditions.yml (hunk 6 above)
Comment (KevinFairise2) [reply .gitlab/deploy_containers/conditions.yml:4]: I replicated the existing rules, looks like job having this rule should not be triggered on nightly

File: .gitlab/deploy_containers/conditions.yml (hunk 5 above)
Comment (KevinFairise2) [reply .gitlab/deploy_containers/conditions.yml:3]: I can update it yes

**Pull request merged**